### PR TITLE
Import the CCT Dev Book

### PR DESCRIPTION
**Pull request by jmtd**
Converted to AsciiDoc and proof-read

### CODE REVIEW

File: docs/devbook.adoc [hunk 1]
```diff
@@ -0,0 +1,267 @@
+= CCT
+David Becvarik, Jonathan Dowland
+:toc: preamble
```
Comment (goldmann) [inline docs/devbook.adoc]: TOC is rendered in wrong place, see preview.

File: docs/devbook.adoc [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+Script modules are the most straightforward way to use CCT. It enables
+you to use your existing bash scripts as CCT modules with the possibility of
+continuously enhancing them with CCT perks (dependencies, artifacts, etc.)
+
+See the https://github.com/jboss-openshift/os-jolokia[os-jolokia] module as an
+example of a script module. You can see it contains the following `module.yaml`
+that introduces the module and its artifacts to CCT:
+
+----
```
Comment (goldmann) [inline docs/devbook.adoc]: Add yaml syntax highlighting.

File: docs/devbook.adoc [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+This is important and shows you how you can group similar
+functionalities to packages inside the module. This enables you to have
+multiple ways of installing jolokia inside the JBoss OpenShift images
+(which is slightly different for every product family).
+
+If you look for example inside the *eap* directory you will find a script called
+https://github.com/dbecvarik/os-jolokia/blob/master/eap/install[install].
+This is executed by the following CCT change:
+
+----
```
Comment (goldmann) [inline docs/devbook.adoc]: Add yaml syntax highlighting.

File: docs/devbook.adoc [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+* Branch
+* Commit id
+
+The preferred way to version modules is to use branches. Each module should
+provide a development branch.
+
+The required module version can be requested via the *version* keyword in the 
+change section of a YAML file as follows:
+
+----
```
Comment (goldmann) [inline docs/devbook.adoc]: Add yaml syntax highlighting.

File: docs/devbook.adoc [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+Artifacts
+~~~~~~~~~
+
+In CCT every module can define *artifacts*. Artifacts are a resource which the
+module needs to use to successfully introduce its features. You can imagine
+an artifact as a JAR file, installer or any other resource which is too
+big to be committed and versioned in the module's Git repository. Artifacts
+are defined in the *artifacts* section of module descriptor.
+
+----
```
Comment (goldmann) [inline docs/devbook.adoc]: Add yaml syntax highlighting.

File: docs/devbook.adoc [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+        - url: https://github.com/jboss-openshift/os-jolokia
+      - changes:
+         - os-jolokia.java:
+            - install:
+----
+
+Module development best practices
+---------------------------------
+
+1.  Create modules around artifacts
```
Comment (goldmann) [inline docs/devbook.adoc:274]: @dbecvarik This needs to be extended and explained why.

Comment (goldmann) [reply docs/devbook.adoc:274]: Not necessarily now, just a note.

File: docs/devbook.adoc [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+   `os-jolokia` module (the directory containing `module.yaml`)
+
+Dogen integration
+~~~~~~~~~~~~~~~~~
+
+Dogen now supports new versions of CCT with bash modules support. To make
+CCT and Dogen talk together you need to place a "cct" configuration section
+in your Dogen `image.yaml` in the following format:
+
+----
```
Comment (goldmann) [inline docs/devbook.adoc]: Add yaml syntax highlighting.

**Comment by jmtd** [comment]
thanks for the review; points addressed.

**Pull request merged**